astroid/interpreter/objectmodel.py

```diff
@@ -100,6 +101,7 @@ def __get__(self, instance, cls=None):
     def __contains__(self, name):
         return name in self.attributes()
 
+    @lru_cache()  # noqa
```

Comment:
Are we sure we want to re-add caching here? @jacobtylerwalls Do you have any data that this saves much time?

Reply:
Yes. :D

When I add a print statement before the clearing of the three LRU caches, and I run:
```python
lint.Run(["pylint"], do_exit=False)
```
I get:
```python
CacheInfo(hits=149138, misses=43009, maxsize=128, currsize=128)  # LookupMixIn.lookup()
CacheInfo(hits=4082, misses=3, maxsize=128, currsize=3)  # _cache_normalize_path_
CacheInfo(hits=75230, misses=13, maxsize=128, currsize=13)  # ObjectModel.attributes()
```

So a realistic benchmark for ObjectModel.attributes would seems to involve ... 7500 hits? Let's try it.

```python
from timeit import timeit
from astroid.interpreter.objectmodel import FunctionModel
fm = FunctionModel()
timeit(fm.attributes, globals=globals(), number=7500)
```
with caching
0.000958
without
0.210987

Reply:
I tested this with `pylint astroid`.

With `lru_cache` I get:
`CacheInfo(hits=77478, misses=26368, maxsize=128, currsize=128)`
With `maxsize=None` I get:
`CacheInfo(hits=83447, misses=20398, maxsize=None, currsize=20398)`

That's not a lot of hits... But it might be worth increasing it a little?

Reply:
How does a 256 cache size fair ? Otherwise, we're already covering 92% of the possible hits with a cache size of 128, it seems we've reached a nice compromise between speed and low memory footprint.

Reply:
I think we can leave it alone, especially because it's an instance method.